src/activate/activate.c

```diff
@@ -201,16 +163,21 @@ static int exec_process(const char *name, char **argv, int start_fd, size_t n_fd
                         safe_close(start_fd);
                 }
 
-                if (asprintf((char **) (envp + n_env++), "LISTEN_FDS=%zu", n_fds) < 0)
+                if (asprintf(&t, "LISTEN_FDS=%zu", n_fds) < 0)
                         return log_oom();
+                r = strv_consume(&envp, TAKE_PTR(t));
+                if (r < 0)
+                        return r;
 
-                if (asprintf((char **) (envp + n_env++), "LISTEN_PID=" PID_FMT, getpid_cached()) < 0)
+                if (asprintf(&t, "LISTEN_PID=" PID_FMT, getpid_cached()) < 0)
                         return log_oom();
+                r = strv_consume(&envp, TAKE_PTR(t));
+                if (r < 0)
+                        return r;
```

Comment:
i wonder if strv_append_printf() would be a helper that could not just shorten this code but a lot more?

Reply:
You mean `strv_extendf()`? I already started writing the function header before I realized it's already there ;)

Reply:
oh, indeed